### PR TITLE
fix(radio,toggle,checkbox,select): padded space is clickable in items

### LINKED ISSUE #27169: bug: form controls in item are not interactive on start/end
### Prerequisites

- [X] I have read the [Contributing Guidelines](https://github.com/ionic-team/ionic-framework/blob/main/.github/CONTRIBUTING.md#creating-an-issue).
- [X] I agree to follow the [Code of Conduct](https://ionicframework.com/code-of-conduct).
- [X] I have searched for [existing issues](https://github.com/ionic-team/ionic-framework/issues) that already report this problem, without success.

### Ionic Framework Version

v7.x

### Current Behavior

Item adds margin to the start and end of the line which means that clicking those spaces causes form controls to not be interactive. In Ionic 6 we had a CSS hack to create an interactive element inside of the form control that covers the entire item: https://github.com/ionic-team/ionic-framework/blob/cb8f07c5530ffc222580a3e3bae8dc85f62c73e5/core/src/components/radio/radio.scss#L65-L67

This resolves the issue for the end margin, but the start margin still was not interactive even in Ionic 6.

### Expected Behavior

The checkbox should get checked.

### Steps to Reproduce

1. Open code reproduction.
2. Click to the left of the label or to the right of the checkbox. Observe that the checkbox does not get checked.

### Code Reproduction URL

https://codepen.io/liamdebeasi/pen/wvYKaEm

### Ionic Info

N/A

### Additional Information

_No response_

### PR DESCRIPTION
**Pull request by sean-perkins**
Issue number: Resolves #27169

---------

## What is the current behavior?

Clicking the padded space within an `ion-item` will not pass the click event to the slotted `ion-radio`, `ion-checkbox`, `ion-select` or `ion-toggle`.

## What is the new behavior?

- The padded space at the start of `.item-native` and at the end of `.item-inner` is clickable to activate a control.
  - When the item is clicked, we check if the event is a result of clicking the control or clicking the item's padded space. If the click event is on the control, we don't need to do anything and let the default behavior occur. If the click event is on the padded space, we manually call the `.click()` method for the interactive element. 
- The cursor pointer displays when hovering over the padded space when a slotted interactive control is present.

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: core/src/components/item/item.tsx [hunk 1]
```diff
@@ -350,6 +350,22 @@ export class Item implements ComponentInterface, AnchorInterface, ButtonInterfac
...
+      /**
+       * Pseudo selectors can't be tested in unit tests.
+       * It will cause an error when running the tests.
+       *
+       * We can remove this once https://github.com/ionic-team/stencil/issues/3588 is fixed.
+       */
+      return undefined;
+    }
+    const controls = this.el.querySelectorAll<HTMLElement>(
+      'ion-toggle:not([disabled]), ion-checkbox:not([disabled]), ion-radio:not([disabled])'
```
Comment (amandaejohnston) [inline core/src/components/item/item.tsx:364]: Should we also do this for `ion-select`? I'm observing this behavior there too, using the `select/item` test.

File: core/src/components/toggle/test/item/toggle.e2e.ts
[the remark this answers is not in the document]
Comment (sean-perkins) [reply core/src/components/toggle/test/item/toggle.e2e.ts:1]: Discussed internally, I'm going to add a comment to the test that describes this behavior is intentional (clicking label activates control).

File: core/src/components/item/item.tsx (hunk 1 above)
Comment (sean-perkins) [reply core/src/components/item/item.tsx:364]: Good catch! Yes, I think we should. I'll update the design doc & then make the changes here.

**Review by amandaejohnston — APPROVED** [review]

**Review by brandyscarney — APPROVED** [review]
I tested this out in a lot of different kinds of items and it is looking good. The added e2e tests look good too. Nice job!

File: core/src/components/item/item.tsx [hunk 2]
```diff
@@ -350,6 +350,22 @@ export class Item implements ComponentInterface, AnchorInterface, ButtonInterfac
     }
   }
 
+  private getFirstInteractive() {
+    if (Build.isTesting) {
+      /**
+       * Pseudo selectors can't be tested in unit tests.
+       * It will cause an error when running the tests.
+       *
+       * We can remove this once https://github.com/ionic-team/stencil/issues/3588 is fixed.
```
[the remark this answers is not in the document]
Comment (sean-perkins) [reply core/src/components/item/item.tsx:359]: Yes I can! The chances of this getting fixed are extremely low though. The root issue comes from Sizzle, a library we don't maintain. Sizzle has been deprecated and moved into jQuery. Stencil will have to add jQuery as a dependency and write a rollup plugin to eliminate all the jQuery-specific code from the bundle.

**Pull request merged**